flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/tier/remote/RemoteStorageScanner.java

```diff
@@ -236,15 +254,24 @@ private boolean checkSegmentExist(
                         partitionId,
                         subpartitionId.getSubpartitionId(),
                         segmentId);
-        boolean isExist;
+        boolean isExist = false;
         try {
             isExist = remoteFileSystem.exists(segmentPath);
-        } catch (IOException e) {
-            throw new RuntimeException("Failed to check segment file. " + segmentPath, e);
+            currentRetryTime = 0;
+        } catch (Throwable t) {
```

Comment:
The type of exceptions is various when the code `remoteFileSystem.exists` is execute because the `remoteFileSystem` can be implemented by OSS, S3, etc. I think using `Throwable` is also acceptable because there is a `MAX_RETRY_TIME`. WDYT? 🤔

Reply:
To be honest, I don't like this way, but I can't think of a better way also. In theory, `FileSystem` should handle (or wrap) exceptions that can be retried by itself, but it doesn't help us do this. 😞